packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts

```diff
@@ -0,0 +1,253 @@
+				required: true,
+			},
+		],
+		outputs: `={{(${configuredOutputs})($parameter)}}`,
+		properties: [
+			{
+				displayName: 'Categories',
+				name: 'categories',
+				placeholder: 'Add Category',
+				type: 'fixedCollection',
+				default: {},
+				typeOptions: {
+					multipleValues: true,
+				},
+				options: [
+					{
+						name: 'categories',
+						displayName: 'Categories',
+						values: [
+							{
+								displayName: 'Category',
+								name: 'category',
+								type: 'string',
+								default: '',
+								description: 'Category to add',
+							},
+							{
+								displayName: 'Description',
+								name: 'description',
+								type: 'string',
+								default: '',
+								description: "Describe your category if it's not obvious",
+							},
+						],
+					},
+				],
+			},
+			{
+				displayName: 'Allow Multiple Classes To Be True',
+				name: 'multiClass',
+				type: 'boolean',
+				default: false,
+			},
+			{
+				displayName: 'Add Fallback Option',
+				name: 'fallback',
+				type: 'boolean',
+				default: false,
+				description: 'Whether to add a "fallback" option if no other categories match',
+			},
+			{
+				displayName: 'Prompt',
+				name: 'promptType',
+				type: 'options',
+				options: [
+					{
+						// eslint-disable-next-line n8n-nodes-base/node-param-display-name-miscased
+						name: 'Take from previous node automatically',
+						value: 'auto',
+						description: 'Looks for an input field called chatInput',
+					},
+					{
+						// eslint-disable-next-line n8n-nodes-base/node-param-display-name-miscased
+						name: 'Define below',
+						value: 'define',
+						description:
+							'Use an expression to reference data in previous nodes or enter static text',
+					},
+				],
+				default: 'auto',
+			},
+			{
+				displayName: 'Text',
+				name: 'text',
+				type: 'string',
+				required: true,
+				default: '',
+				placeholder: 'e.g. Hello, how can you help me?',
+				typeOptions: {
+					rows: 2,
+				},
+				displayOptions: {
+					show: {
+						promptType: ['define'],
+					},
+				},
+			},
+			{
+				displayName: 'Options',
+				name: 'options',
+				type: 'collection',
+				default: {},
+				placeholder: 'Add Option',
+				options: [
+					{
+						displayName: 'System Prompt Template',
+						name: 'systemPromptTemplate',
+						type: 'string',
+						default: SYSTEM_PROMPT_TEMPLATE,
+						description: 'String to use directly as the system prompt template',
+						typeOptions: {
+							rows: 6,
+						},
+					},
+				],
+			},
+		],
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+
+		const llm = (await this.getInputConnectionData(
+			NodeConnectionType.AiLanguageModel,
+			0,
+		)) as BaseLanguageModel;
+
+		const categories = (
+			this.getNodeParameter('categories', 0) as {
+				categories: [{ category: string; description: string }];
+			}
+		)?.categories;
```

Comment:
```suggestion
const categories = this.getNodeParameter('categories.categories', 0) as Array<{
			category: string;
			description: string;
		}>;
```

Reply:
ah, you can request sub-parameters, thx for that!